kolibri/plugins/setup_wizard/assets/src/views/ImportIndividualUserForm.vue

```diff
@@ -151,7 +153,32 @@
         return this.$tr('enterAdminCredentials', { facility: this.facility.name });
       },
     },
+    beforeMount() {
+      this.fetchNetworkLocation(this.deviceId).then(() => {
+        if (!this.facility) {
+          this.$store.dispatch('showError', 'Failed to retrieve facilities.');
+        }
+      });
+    },
     methods: {
+      fetchNetworkLocation(deviceId) {
+        this.loadingNewAddress = true;
+        return this.fetchNetworkLocationFacilities(deviceId)
+          .then(data => {
+            this.facilities = [...data.facilities];
+            this.device = {
+              name: data.device_name,
+              id: data.device_id,
+              baseurl: data.device_address,
+            };
+            this.selectedFacilityId = this.facilities[0].id;
+            this.loadingNewAddress = false;
+          })
+          .catch(error => {
+            // TODO handle disconnected peers error more gracefully
+            this.$store.dispatch('showError', error);
```

Comment:
Looks like `fetchNetworkLocationFacilities` already catches, and incorrectly because it returns an array, but `{facilities: []}` would be expected.

Reply:
This will be resolved with 'Handle `apiError`...'? It's dead code because of https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/mixins/commonSyncElements.js#L77